hera_sim/tests/test_simulator.py

```diff
@@ -149,6 +149,30 @@ def __call__(self, lsts, freqs):
         assert np.all(base_sim.data.data_array == 1)
 
 
+def test_add_with_full_array_return(base_sim):
+    @component
+    class TestBase:
+        pass
+
+    class Test(TestBase):
+        return_type = "full_array"
+        attrs_to_pull = dict(
+            freqs="freqs",
+            ant_1_array="ant_1_array",
+            pols="polarization_array",
+        )
```

Comment:
Sure, but I think I'd need to think a little carefully to do it right. Are you thinking of doing that so we can omit things from `attrs_to_pull`? Like, instead of the above, we would just do `attrs_to_pull = {"pols": "polarization_array"}`?

Reply:
OK, I've implemented it now. Relevant changes are in the `Simulator._initialize_args_from_model` method (in particular, lines 932-935).